Foundation/NSObjCRuntime.swift

```diff
@@ -114,3 +242,106 @@ internal protocol _NSBridgable {
+ */
+public func NSStringFromClass(aClass: AnyClass) -> String {
+    let demangledName = _typeName(aClass).bridge()
+    let components = demangledName.componentsSeparatedByString(".")
+    
+    if components.count == 2 {
+        if components[0] == _SwiftFoundationModuleName {
+            return components[1]
+        } else {
+            return String(demangledName)
+        }
+    } else if let mangledName = _CFCopyNominalTypeNameForClass(unsafeBitCast(aClass, CFTypeRef.self)) {
+        return "_Tt" + String(mangledName)
+    } else {
+        fatalError("NSStringFromClass could not determine class name")
+    }
+}
+
+#if os(OSX) || os(iOS)
+    private let RTLD_DEFAULT = UnsafeMutablePointer<Void>(bitPattern: -2)
+    typealias TypeMetadataAccessor = @convention(c) () -> AnyClass?
+#elseif os(Linux)
+    // XXX @convention(c) crashes swiftc
+    typealias TypeMetadataAccessor = () -> AnyClass?
+    private let RTLD_DEFAULT = UnsafeMutablePointer<Void>(bitPattern: 0)
+#endif
+
+/**
+    Calls a metadata accessor given a metadata accessor symbol name.
+ */
+private func typeMetadataFromAccessorName(mangledName : String) -> AnyClass? {
+    let symbol = dlsym(RTLD_DEFAULT, mangledName)
+    
+    if symbol != nil {
+        let accessor = unsafeBitCast(symbol, TypeMetadataAccessor.self)
+        return accessor()
+    } else {
+        return nil
+    }
+}
+
+/**
+    Returns mangled metadata accessor symbol name for a namespaced
+    Swift class.
+ */
+private func mangledTypeNameNameForClass(components : [String]) -> String {
+    var mangledName = "_TtC"
+    
+    for component in components {
+        mangledName += "\(component.length)" + component
+    }
+    
+    return mangledName
+}
+
+private func mangledTypeNameNameForClass(className : String) -> String {
+    let components = className.bridge().componentsSeparatedByString(".")
+    
+    return mangledTypeNameNameForClass(components)
+}
+
+/**
+    Returns the class metadata given a string, applying the following rules:
+ 
+    - Mangled type names are looked up directly
+    - Unmangled namespaced names are mangled then looked up
+    - Unmangled unnamespaced names are mangled into SwiftFoundation classes before lookup
+ */
+public func NSClassFromString(aClassName: String) -> AnyClass? {
+    var mangledName : String
+    
+    if aClassName.hasPrefix("_Tt") {
+        mangledName = aClassName
+    } else {
+        let components = aClassName.bridge().componentsSeparatedByString(".")
+        
+        if components.count > 1 {
+            mangledName = mangledTypeNameNameForClass(aClassName)
+        } else {
+            mangledName = mangledTypeNameNameForClass([_SwiftFoundationModuleName] + components)
+        }
+    }
+    
+    if !mangledName.hasPrefix("_Tt") {
+        fatalError("NSClassFromString() invalid mangled name \(mangledName)")
+    }
+
+    let accessorName = "_TMa" + mangledName.bridge().substringFromIndex(3)
```

Comment:
We really need an official method to do this from the runtime. Lets create a few feature request bugs to track that and create a few bugs to improve this to be less fragile and block them on that. Since the start of this project we have already seen the mangler change once so it is reasonable to think that the mangled names may change again.